### PR TITLE
Feature/mx 1596 simplify settings handling

### PR DESCRIPTION
**Pull request by rababerladuseladim**
# PR Context

depends on: 
- https://github.com/robert-koch-institut/mex-common/pull/220
- mex-assets PR 169

# Changes

- BREAKING: integrate extractor specific settings in main extractor settings class. 
  Environment variables change from `EXTRACTOR_PARAMETER` to `MEX_EXTRACTOR__PARAMETER`,
  access from `ExtractorSettings.parameter` to `settings.extractor.parameter`.
- update mex-common to 0.32.0

### CODE REVIEW

File: mex/international_projects/main.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def extracted_international_projects_activities(
     return mex_sources
 
 
-@entrypoint(InternationalProjectsSettings)
+@entrypoint(Settings)
```
Comment (mr-kamran-ali) [inline mex/international_projects/main.py:132]: How does dagster figures out the correct settings for specific extractor? just by the variable name `international_projects` in `Settings`?

File: mex/artificial/settings.py [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
+from mex.common.models import BaseModel
 from mex.common.types import AssetsPath
-from mex.settings import Settings
 
 
-class ArtificialSettings(Settings):
-    """Artificial settings definition for the artificial data creator."""
-
-    model_config = SettingsConfigDict(env_prefix="artificial_")
+class ArtificialSettings(BaseModel):
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply mex/artificial/settings.py:7]: This PR unifies all extractors settings in a single class: `mex.settings.Settings`. The motivation for that can be found in the ticket but in short synchronization of shared settings (e.g. `assets_dir`) across different settings instances was unreliable. Hence we wanted to get rid of shared settings. My chosen route is to unite all settings in a single class. I made all the extractor specific settings submodules of the main extractors settings, according to https://docs.pydantic.dev/latest/concepts/pydantic_settings/#usage

File: mex/biospecimen/settings.py [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
 
+from mex.common.models import BaseModel
 from mex.common.types import AssetsPath
-from mex.settings import Settings
 
 
-class BiospecimenSettings(Settings):
-    """Settings for the Biospecimen extractor."""
-
-    model_config = SettingsConfigDict(env_prefix="biospecimen_")
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply mex/biospecimen/settings.py:11]: Pydantic builds them. `mex.common.settings.BaseSettings` sets `env_prefix` to `mex_` and `env_nested_delimiter` (how submodel attributes are separated from top-level attributes) to `__`. See pydantic docs for further info: https://docs.pydantic.dev/latest/concepts/pydantic_settings/#parsing-environment-variable-values

File: mex/international_projects/main.py (hunk 1 above)
Comment (rababerladuseladim) [reply mex/international_projects/main.py:132]: There is only one setting: `mex.settings.Settings`. Nothing to figure out for dagster.

**Review by mr-kamran-ali — APPROVED** [review]
🚀

**Pull request merged**